planner/core/physical_plans.go

```diff
@@ -915,11 +915,27 @@ type PhysicalUnionAll struct {
 	physicalSchemaProducer
 }
 
+// AggMppRunMode defines the running mode of aggregation in MPP
+type AggMppRunMode int
+
+const (
+	// NoMpp means the default value which runs not in MPP
+	NoMpp AggMppRunMode = iota
+	// Mpp1Phase runs only 1 phase but requires its child's partition property
+	Mpp1Phase
+	// Mpp2Phase runs partial agg + final agg with hash partition
+	Mpp2Phase
+	// MppTiDB runs agg on TiDB (and a partial agg on TiFlash if in 2 phase agg)
+	MppTiDB
+)
+
 type basePhysicalAgg struct {
 	physicalSchemaProducer
 
-	AggFuncs     []*aggregation.AggFuncDesc
-	GroupByItems []expression.Expression
+	AggFuncs      []*aggregation.AggFuncDesc
+	GroupByItems  []expression.Expression
+	MppRunMode    AggMppRunMode
+	PartitionCols []*expression.Column
```

Comment:
The keyword `partition` is ambiguous since it is used by two main features in TiDB(`TablePartition` and `MPP`).
I advise adding a prefix `mpp` when using it in MPP.

Reply:
renamed.